core/control-plane/control-plane-transfer/src/main/java/org/eclipse/edc/connector/controlplane/transfer/process/TransferProcessManagerImpl.java

```diff
@@ -177,6 +180,7 @@ protected StateMachineManager.Builder configureStateMachineManager(StateMachineM
                 .processor(processConsumerTransfersInState(RESUMING, this::processConsumerResuming))
                 .processor(processTransfersInState(COMPLETING, this::processCompleting))
                 .processor(processTransfersInState(TERMINATING, this::processTerminating))
+                .processor(processTransfersInState(TERMINATING_REQUESTED, this::processTerminating))
```

Comment:
If the state `TERMINATING_REQUESTED` is handled by `processTerminating`, wouldn't that mean that a `TransferTerminationMessage` is sent again after one has already been sent by the counter-party? Afaik, only one party should send this message, right?

Reply:
yes, that's why there's a condition into the `sendTransferTerminationMessage`.
I thought to bring the condition to the upper lever, but it would had complicated the code much more, because of the nested `RetryProcess` used. I can take a look if it could refactored into a better shape

Reply:
You're right, I completely missed that. While it reads a bit intransparent at first, if the alternative is refactoring it all in a more compley way, this would be fine with me.